### ISSUE TITLE
cilium hubble commands not working when ipv6 is disabled on host

### ISSUE THREAD

**Issue by jr200** [cilium/cilium-cli#1093]
## Bug report

**General Information**

- Cilium CLI version (run `cilium version`)

```
cilium-cli: v0.12.3 compiled with go1.19 on linux/amd64
cilium image (default): v1.12.1
cilium image (stable): v1.12.2
cilium image (running): v1.12.2
```

- Orchestration system version in use (e.g. `kubectl version`, ...)

```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.2", GitCommit:"8b5a19147530eaac9476b0ab82980b4088bbc1b2", GitTreeState:"clean", BuildDate:"2021-09-15T21:38:50Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.2", GitCommit:"8b5a19147530eaac9476b0ab82980b4088bbc1b2", GitTreeState:"clean", BuildDate:"2021-09-15T21:32:41Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
```

- Platform / infrastructure information (e.g. AWS / Azure / GCP, image / kernel versions)

```
Linux sd1b-jrag01 5.4.0-110-generic #124~18.04.1-Ubuntu SMP Fri Apr 22 12:01:21 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**How to reproduce the issue**

1. Disable ipv6 on host.
2. Run `cilium hubble port-forward` or `cilium hubble ui`

```bash
$ cilium hubble port-forward
❌ Unable to execute "kubectl port-forward -n kube-system svc/hubble-relay --address 0.0.0.0 --address :: 4245:80":
 Forwarding from 0.0.0.0:4245 -> 4245
Unable to listen on port 4245: Listeners failed to create with the following errors: [unable to create listener: Error listen tcp6 [::]:4245: socket: address family not supported by protocol]
error: unable to listen on any of the requested ports: [{4245 4245}]

Error: Unable to port forward: unable to execute "kubectl port-forward -n kube-system svc/hubble-relay --address 0.0.0.0 --address :: 4245:80": exit status 1
```

```bash
$ cilium hubble ui
❌ Unable to execute "kubectl port-forward -n kube-system svc/hubble-ui --address 0.0.0.0 --address :: 12000:80":
 Forwarding from 0.0.0.0:12000 -> 8081
Unable to listen on port 12000: Listeners failed to create with the following errors: [unable to create listener: Error listen tcp6 [::]:12000: socket: address family not supported by protocol]
error: unable to listen on any of the requested ports: [{12000 8081}]


Error: Unable to port forward: unable to execute "kubectl port-forward -n kube-system svc/hubble-ui --address 0.0.0.0 --address :: 12000:80": exit status 1
```

**Expected behaviour**

These functions should generate the `kubectl port-forward` commands without binding to the ipv6 address.

**Comment by jakeWatsonOakland** [comment]
I get similar while following the tutorial for default EKS install (IPAM=ENI) - this is likely because ipv6 does not work in EKS ENI mode.

**Comment by pr0PM** [comment]
Still facing same issue on similar setup. At this stage hubble ui front-end won't come up as it keeps crashing non stop:

```
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: ipv6 not available
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2023/01/05 18:26:23 [emerg] 1#1: socket() :8081 failed (97: Address family not supported by protocol)
nginx: [emerg] socket() :8081 failed (97: Address family not supported by protocol)
Stream closed EOF for kube-system/hubble-ui-5db65d4679-jgfbf (frontend)
```

@jakeWatsonOakland @jr200 were you able to find any workarounds for this?

**Comment by jr200** [comment]
> Still facing same issue on similar setup. At this stage hubble ui front-end won't come up as it keeps crashing non stop:
> 
> ```
> /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
> /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
> /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
> 10-listen-on-ipv6-by-default.sh: info: ipv6 not available
> /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
> /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
> /docker-entrypoint.sh: Configuration complete; ready for start up
> 2023/01/05 18:26:23 [emerg] 1#1: socket() :8081 failed (97: Address family not supported by protocol)
> nginx: [emerg] socket() :8081 failed (97: Address family not supported by protocol)
> Stream closed EOF for kube-system/hubble-ui-5db65d4679-jgfbf (frontend)
> ```
> 
> @jakeWatsonOakland @jr200 were you able to find any workarounds for this?

Thats a different issue I think, relating to the startup of nginx containers when ipv6 is disabled on the host. I might be wrong, but I think this was fixed by [https://github.com/cilium/cilium/pull/21127](https://github.com/cilium/cilium/pull/21127)? ...what version of cilium are you using?

Before the above release, I was doing this as a workaround after a helm install/upgrade:

```
kubectl get -n kube-system cm hubble-ui-nginx -o yaml | sed s/listen.*::.*[\;]//g |  kubectl apply -f -
```

**Comment by devopstales** [comment]
Set `hubble.ui.frontend.server.ipv6.enabled: false`
